src/command/subgraph/check.rs

```diff
@@ -0,0 +1,140 @@
+use prettytable::{cell, row, Table};
+use serde::Serialize;
+use structopt::StructOpt;
+
+use rover_client::query::subgraph::check;
+
+use crate::client::StudioClientConfig;
+use crate::command::RoverStdout;
+use crate::utils::loaders::load_schema_from_flag;
+use crate::utils::parsers::{parse_graph_ref, parse_schema_source, GraphRef, SchemaSource};
+
+#[derive(Debug, Serialize, StructOpt)]
+pub struct Check {
+    /// <NAME>@<VARIANT> of graph in Apollo Studio to validate.
+    /// @<VARIANT> may be left off, defaulting to @current
+    #[structopt(name = "GRAPH_REF", parse(try_from_str = parse_graph_ref))]
+    #[serde(skip_serializing)]
+    graph: GraphRef,
+
+    /// Name of the implementing service to validate
+    #[structopt(long = "service", required = true)]
+    #[serde(skip_serializing)]
+    service_name: String,
+
+    /// Name of configuration profile to use
+    #[structopt(long = "profile", default_value = "default")]
+    #[serde(skip_serializing)]
+    profile_name: String,
+
+    /// The schema file to push
+    /// Can pass `-` to use stdin instead of a file
+    #[structopt(long, short = "s", parse(try_from_str = parse_schema_source))]
+    #[serde(skip_serializing)]
+    schema: SchemaSource,
+}
+
+impl Check {
+    pub fn run(&self, client_config: StudioClientConfig) -> Result<RoverStdout> {
+        let client = client_config.get_client(&self.profile_name)?;
+
+        let sdl = load_schema_from_flag(&self.schema, std::io::stdin())?;
+
+        let partial_schema = check::check_partial_schema_query::PartialSchemaInput {
+            sdl: Some(sdl),
+            // we never need to send the hash since the back end computes it from SDL
+            hash: None,
+        };
+        let res = check::run(
+            check::check_partial_schema_query::Variables {
+                graph_id: self.graph.name.clone(),
+                variant: self.graph.variant.clone(),
+                partial_schema,
+                implementing_service_name: self.service_name.clone(),
+            },
+            &client,
+        )
+        .context("Failed to validate schema")?;
+
+        tracing::info!(
+            "Validated the proposed subgraph against metrics from {}@{}",
+            &self.graph.name,
+            &self.graph.variant
+        );
+
+        match res {
+            check::CheckResponse::CompositionErrors(composition_errors) => {
+                handle_composition_errors(&composition_errors)
+            }
+            check::CheckResponse::CheckResult(check_result) => handle_checks(check_result),
+        }
+    }
+}
+
+fn handle_checks(check_result: check::CheckResult) -> Result<RoverStdout> {
+    let num_changes = check_result.changes.len();
+
+    let msg = match num_changes {
+        0 => "There is no difference between the proposed subgraph and the subgraph that already exists in the graph registry. Try making a change to your proposed subgraph before running this command.".to_string(),
```

Comment:
We might want to use different wording for the "no changes" dialog, since when studio returns there are "no changes" that doesn't mean the document didn't update and especially for a federated variant It doesn't mean the subgraph is the same. 

When studio runs a check it only looks for a subset of all possible changes to a schema document ignoring whitespace, ordering, and directives. This means that there can be 0 "changes" but things like `@requires` and `@key` might have changed in the subgraph schema which has a meaningful change for federation but is not currently represented in the check diff.

The change list was built around what was important for operation execution so it isn't an exhaustive way to say these schemas are effectively identical.

Reply:
What do you think a better message would be in this case?

Reply:
I've taken a swing at it with `There were no changes detected between the proposed subgraph and the subgraph that already exists in the graph registry.`

Reply:
I think no changes detected between in the composed executable schema? The changes run on a version of the composed schema without the composition directives.